deployer/register.go

```diff
@@ -142,15 +144,28 @@ func Register(g *herd.Graph, artifact schema.ReleaseArtifact, c schema.Config, c
 
 	// Extract SquashFS from released asset to build the raw disk image if needed
 	g.Add(opExtractSquashFS,
-		herd.EnableIf(func() bool { return diskIsSet && !fromImage }),
+		herd.EnableIf(func() bool { return rawDiskIsSet && !fromImage }),
 		herd.WithDeps(opDownloadSquashFS), herd.WithCallback(ops.ExtractSquashFS(squashFSfile, tmpRootfs)))
 
 	imageOrSquashFS := herd.IfElse(fromImage, herd.WithDeps(opContainerPull), herd.WithDeps(opExtractSquashFS))
 
 	g.Add(opGenRawDisk,
-		herd.EnableIf(func() bool { return diskIsSet && c.Disk.ARM == nil }),
+		herd.EnableIf(func() bool { return rawDiskIsSet && c.Disk.ARM == nil && !c.Disk.MBR }),
 		imageOrSquashFS,
-		herd.WithCallback(ops.GenRawDisk(tmpRootfs, filepath.Join(dst, "disk.raw"))))
+		herd.WithCallback(ops.GenEFIRawDisk(tmpRootfs, filepath.Join(dst, "disk.raw"))))
+
+	g.Add(opGenMBRRawDisk,
+		herd.EnableIf(func() bool { return c.Disk.ARM == nil && c.Disk.MBR }),
```

Comment:
why don't we need to validate here `rawDiskIsSet`?

Reply:
raw disks are images which are generated statically, without the need of an hypervisor - here instead we run QEMU to generate an image